src/ngraph/graph_util.hpp

```diff
@@ -292,6 +292,9 @@ namespace ngraph
     // the output of this node with in-place kernels
     bool possibly_overwritten(Node* node);
 
+    // Return true if node has state.
```

Comment:
Let's make this a virtual with a default definition in node that returns false.

Reply:
If we add m_has_state and has_state() to Node, then we don't need has_state here. cpu_external_function could just do node->has_state().